src/plugins/data_source_management/public/plugin.ts

```diff
@@ -303,10 +319,85 @@ export class DataSourceManagementPlugin
     };
     setRenderAssociatedObjectsDetailsFlyout(renderAssociatedObjectsDetailsFlyout);
 
+    const getDirectQuerySyncFromUrl = (): boolean | null => {
+      const parsedHash = parseUrlHash(window.location.href);
+      if (parsedHash && parsedHash.query) {
+        const directQuerySync = parsedHash.query.directQuerySync;
+        if (directQuerySync === 'true') return true;
+        if (directQuerySync === 'false') return false;
+      }
+      return null;
+    };
+
+    const shouldMountBanner = (): boolean => {
+      const urlFlag = getDirectQuerySyncFromUrl();
+      if (urlFlag !== null) {
+        return urlFlag;
+      }
+      return this.config.dashboardDirectQuerySyncEnabled;
+    };
+
+    const updateBanner = () => {
+      const appId = this.currentAppId;
+
+      if (appId === 'dashboards') {
+        const hash = window.location.hash;
+        const isDashboardViewMatch = hash.match(/#\/view\/([^\/?]+)(\?.*)?$/);
+        if (isDashboardViewMatch && isDashboardViewMatch[1]) {
+          const dashboardId = isDashboardViewMatch[1];
+          if (shouldMountBanner()) {
+            if (!this.bannerId) {
+              this.bannerId = core.overlays.banners.add(
+                toMountPoint(
+                  React.createElement(DashboardDirectQuerySyncBanner, {
+                    dashboardId,
+                    http: core.http,
+                    notifications: core.notifications,
+                    savedObjectsClient: core.savedObjects.client,
+                    removeBanner: () => {
+                      if (this.bannerId) {
+                        core.overlays.banners.remove(this.bannerId);
+                        this.bannerId = null;
+                      }
+                    },
+                  })
+                )
+              );
+            }
+          } else if (this.bannerId) {
+            core.overlays.banners.remove(this.bannerId);
+            this.bannerId = null;
```

Comment:
nit: could this logic be simplified  to a single or statement?

Reply:
I was trying to change it to:
```
          this.bannerId && (core.overlays.banners.remove(this.bannerId), (this.bannerId = null));
```
However, the above gives me the linting error:
```
        361:21  error  Expected an assignment or function call and instead saw an expression  @typescript-eslint/no-unused-expressions
```